src/Microsoft.Framework.PackageManager/Restore/LockFileFormat.cs

```diff
@@ -20,12 +21,37 @@ public class LockFileFormat
 
         public LockFile Read(string filePath)
         {
-            using (var stream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read))
+            using (var stream = OpenFileStream(filePath))
             {
                 return Read(stream);
             }
         }
 
+        private static FileStream OpenFileStream(string filePath)
+        {
+            int retry = 3;
+            while (true)
+            {
+                try
+                {
+                    return new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read);
+                }
+                catch (Exception ex)
```

Comment:
We want to resolve the lockfile read/write conflicts in this PR. However, this will retry even when some other kinds of exceptions are thrown. Is it possible that we narrow down to a more specific exception?